include/envoy/network/listener.h

```diff
@@ -134,7 +134,9 @@ struct UdpRecvData {
  * Encapsulates the information needed to send a udp packet to a target
  */
 struct UdpSendData {
-  Address::InstanceConstSharedPtr send_address_;
+  const Address::Ip* local_ip_;
+  const Address::InstanceConstSharedPtr& peer_address_;
```

Comment:
nit: storing a reference to the shared ptr like this is IMO a bit shady, can you just store the reference to the underlying object assuming it should not be null? Or use a raw pointer?

Reply:
Changed to Address::Instance&